questions.js

```diff
@@ -442,3 +442,208 @@ db.restaurants.find({
     { borough: { $in: ["Manhattan", "Brooklyn"] } },
   ],
 });
+
+// 41.Write a MongoDB query to find the restaurants that have a grade with a score of 2 and
+// a grade with a score of 6 and are located in the borough of Manhattan or Brooklyn, and
+// their cuisine is not American.
+
+db.restaurants.find({
+  $and: [
+    { "grades.score": { $eq: 2 } },
+    { "grades.score": { $eq: 6 } },
+    { borough: { $in: ["Manhattan", "Brooklyn"] } },
+    { cuisine: { $ne: "American " } },
+  ],
+});
+
+// 42. Write a MongoDB query to find the restaurants that have a grade with a score of 2
+// and a grade with a score of 6 and are located in the borough of Manhattan or Brooklyn,
+// and their cuisine is not American or Chinese.
+
+db.restaurants.find({
+  $and: [
+    { "grades.score": { $eq: 2 } },
+    { "grades.score": { $eq: 6 } },
+    { borough: { $in: ["Manhattan", "Brooklyn"] } },
+    { cuisine: { $nin: ["American ", "Chinese"] } },
+  ],
+});
+
+// 43. Write a MongoDB query to find the restaurants that have a grade with a score of
+// 2 or a grade with a score of 6.
+
+db.restaurants.find({ "grades.score": { $in: [2, 6] } });
+
+// 44. Write a MongoDB query to find the restaurants that have a grade with a score
+// of 2 or a grade with a score of 6 and are located in the borough of Manhattan.
+
+db.restaurants.find({
+  $and: [{ "grades.score": { $in: [2, 6] } }, { borough: "Manhattan" }],
+});
+
+// 45. Write a MongoDB query to find the restaurants that have a grade with a score
+// of 2 or a grade with a score of 6 and are located in the borough of Manhattan or Brooklyn.
+
+db.restaurants.find({
+  $and: [
+    { "grades.score": { $in: [2, 6] } },
+    { borough: { $in: ["Manhattan", "Brooklyn"] } },
+  ],
+});
+
+// 46. Write a MongoDB query to find the restaurants that have a grade with a score
+// of 2 or a grade with a score of 6 and are located in the borough of Manhattan
+// or Brooklyn, and their cuisine is not American.
+
+db.restaurants.find({
+  $and: [
+    { "grade.score": { $in: [2, 6] } },
+    { Borough: { $in: ["Manhattan", "Brooklyn"] } },
+    { Cuisine: { $ne: "American " } },
+  ],
+});
+
+// 47. Write a MongoDB query to find the restaurants that have a grade with a
+// score of 2 or a grade with a score of 6 and are located in the borough of
+// Manhattan or Brooklyn, and their cuisine is not American or Chinese.
+
+db.restaurants.find({
+  $and: [
+    { "grade.score": { $in: [2, 6] } },
+    { Borough: { $in: ["Manhattan", "Brooklyn"] } },
+    { Cuisine: { $ne: "American ", $ne: "Chinese" } },
+  ],
+});
+
+// 48. Write a MongoDB query to find the restaurants that have all grades with a score greater than 5.
+
+db.restaurants.find({
+  grades: { $not: { $elemMatch: { score: { $lte: 5 } } } },
```

Comment:
why not just write $gte 6
why have you written it as a not condition here.

Reply:
As grades is an array of objects , if i use $gte 6 , it will return the document even if one score is 2 and another is 8, but here we require all the score to be more than 5, that's why i am using not condition as it will reject if any value is <=5